src/Controller/Plugin/Forward.php

```diff
@@ -193,7 +193,7 @@ protected function detachProblemListeners(SharedEvents $sharedEvents)
 
                     foreach ($classArray as $class) {
                         if ($currentCallback instanceof $class) {
-                            $sharedEvents->detach($id, $currentEvent);
+                            $sharedEvents->detach($currentEvent, $id);
```

Comment:
It's a rarely used feature, TBH; there are very few use cases currently for detaching listeners, much less shared listeners, and the `Forward` plugin here is literally the only place I've seen it in the ZF2 or Apigility code bases.

Though with React, I suspect it will becomes more common. Those users will adopt v3 anyways, as it's faster. :smile: